### ISSUE TITLE
Can't install i3xrocks-time

### ISSUE THREAD

**Issue by MaestroGlanz** [regolith-linux/regolith-i3xrocks-config#144]
I'm hopefully at the right point here:
I updated to the latest packages, especially all regolith packages. At first, the clock did work, now I get an error, when trying to update it:

```
sudo apt install i3xrocks-time 
Paketlisten werden gelesen… Fertig
Abhängigkeitsbaum wird aufgebaut… Fertig
Statusinformationen werden eingelesen… Fertig
Einige Pakete konnten nicht installiert werden. Das kann bedeuten, dass
Sie eine unmögliche Situation angefordert haben oder, wenn Sie die
Unstable-Distribution verwenden, dass einige erforderliche Pakete noch
nicht erstellt wurden oder Incoming noch nicht verlassen haben.
Die folgenden Informationen helfen Ihnen vielleicht, die Situation zu lösen:

Die folgenden Pakete haben unerfüllte Abhängigkeiten:
 regolith-control-center : Hängt ab von: libadwaita-1-0 (>= 1.2~alpha) aber 1.1.0-1ubuntu2 soll installiert werden
                           Hängt ab von: libcolord-gtk4-1 (>= 0.1.24) ist aber nicht installierbar
                           Hängt ab von: libgnome-bg-4-2 (>= 3.27.90) ist aber nicht installierbar
                           Hängt ab von: libgnome-desktop-4-2 (>= 3.33.4) ist aber nicht installierbar
                           Hängt ab von: libgnome-rr-4-2 (>= 3.17.92) ist aber nicht installierbar
                           Hängt ab von: libgoa-1.0-0b (>= 3.45) aber 3.44.0-1ubuntu1 soll installiert werden
                           Hängt ab von: libnma-gtk4-0 (>= 1.8.34) ist aber nicht installierbar
                           Hängt ab von: gnome-control-center-data (>= 1:43.0) aber 1:41.7-0ubuntu0.22.04.5 soll installiert werden
                           Empfiehlt: gnome-bluetooth-sendto ist aber nicht installierbar
                           Empfiehlt: gnome-user-share soll aber nicht installiert werden
                           Empfiehlt: libcanberra-pulse soll aber nicht installiert werden
                           Empfiehlt: malcontent-gui soll aber nicht installiert werden
                           Empfiehlt: libnss-myhostname soll aber nicht installiert werden
E: Probleme können nicht korrigiert werden, Sie haben zurückgehaltene defekte Pakete.

```

Which translates to:

```
sudo apt install i3xrocks-time
Reading package lists... Done
Building dependency tree... Done
Status information is read in... Done
Some packages could not be installed. That can mean that
You have requested an impossible situation or if you are the
Unstable distribution that still use some required packages
have not been created or have not yet left Incoming.
The following information may help you to resolve the situation:

The following packages have unmet dependencies:
  regolith-control-center : Depends on: libadwaita-1-0 (>= 1.2~alpha) but 1.1.0-1ubuntu2 should be installed
                            Depends on: libcolored-gtk4-1 (>= 0.1.24) but is not installable
                            Depends on: libgnome-bg-4-2 (>= 3.27.90) but is not installable
                            Depends on: libgnome-desktop-4-2 (>= 3.33.4) but is not installable
                            Depends on: libgnome-rr-4-2 (>= 3.17.92) but is not installable
                            Depends on: libgoa-1.0-0b (>= 3.45) but 3.44.0-1ubuntu1 should be installed
                            Depends on: libnma-gtk4-0 (>= 1.8.34) but is not installable
                            Depends on: gnome-control-center-data (>= 1:43.0) but 1:41.7-0ubuntu0.22.04.5 is about to be installed
                            Recommends: gnome-bluetooth-sendto but cannot be installed
                            Recommends: gnome-user-share should not be installed
                            Recommends: libcanberra-pulse should not be installed
                            Recommends: malcontent-gui should not be installed
                            Recommends: libnss-myhostname should not be installed
E: Unable to correct problems, you have held defective packages.

```

I'm not sure, if I can fix the problem, or there must be a correction in the dependencies of the package.

**Comment by kgilmer** [comment]
Hmm, I'm guessing that there is a mismatch between the host Ubuntu version and the Regolith version you installed.  What does `lsb_release -a` and `cat /etc/apt/sources.list.d/regolith.list` return?

**Comment by MaestroGlanz** [comment]
```
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.1 LTS
Release:	22.04
Codename:	jammy

```
```
deb [arch=amd64 signed-by=/usr/share/keyrings/regolith-archive-keyring.gpg] https://regolith-desktop.org/release-ubuntu-kinetic-amd64 kinetic main
```

cat * results in 
```
deb https://ppa.launchpadcontent.net/apandada1/brightness-controller/ubuntu/ jammy main
deb http://ppa.launchpad.net/blaze/rtbth-dkms/ubuntu/ groovy main
deb https://ppa.launchpadcontent.net/cdemu/ppa/ubuntu/ jammy main
deb [signed-by=/usr/share/keyrings/ni-software-2022-focal.asc] https://download.ni.com/ni-linux-desktop/2022/Q4/deb/ni/focal focal ni
deb [signed-by=/usr/share/keyrings/ni-software-2022-jammy.asc] https://download.ni.com/ni-linux-desktop/2022/Q4/deb/ni/jammy jammy ni
deb [signed-by=/usr/share/keyrings/ni-software-stream-focal.asc] https://download.ni.com/ni-linux-desktop/stream/deb/ni/focal focal ni
deb [signed-by=/usr/share/keyrings/ni-software-stream-jammy.asc] https://download.ni.com/ni-linux-desktop/stream/deb/ni/jammy jammy ni
deb https://ppa.launchpadcontent.net/ozmartian/apps/ubuntu/ jammy main
deb [arch=amd64 signed-by=/usr/share/keyrings/regolith-archive-keyring.gpg] https://regolith-desktop.org/release-ubuntu-kinetic-amd64 kinetic main
deb http://ppa.launchpad.net/sicklylife/gnucash/ubuntu/ groovy main
deb http://ppa.launchpad.net/system76/pop/ubuntu/ groovy main
deb [arch=amd64] https://packages.microsoft.com/repos/ms-teams stable main
Types: deb
URIs: https://dl.winehq.org/wine-builds/ubuntu
Suites: jammy
Components: main
Architectures: amd64 i386
Signed-By: /etc/apt/keyrings/winehq-archive.key

```

Might it be caused by the groovy ones?

**Comment by MaestroGlanz** [comment]
I removed the groovy things, still no change.

**Comment by MaestroGlanz** [comment]
I removed regolith-desktop and ubuntu-desktop, made autoremove and installed both again. No change.

**Comment by MaestroGlanz** [comment]
@kgilmer 
I found the problem. On regolith-desktop.com it says in the installation instruction

```
echo deb "[arch=amd64 signed-by=/usr/share/keyrings/regolith-archive-keyring.gpg] \
https://regolith-desktop.org/release-ubuntu-kinetic-amd64 kinetic main" | \
sudo tee /etc/apt/sources.list.d/regolith.list
```
But I have jammy.

The instruction should be like this:
```
CURR_U_NAME=$(lsb_release -a | grep Codename | grep -e ":\s*\w*" -o | grep -e "\w*" -o)
echo deb "[arch=amd64 signed-by=/usr/share/keyrings/regolith-archive-keyring.gpg] \
https://regolith-desktop.org/release-ubuntu-$CURR_U_NAME-amd64 $CURR_U_NAME main" | \
sudo tee /etc/apt/sources.list.d/regolith.list
```
Therefore it will be safe for different Ubuntu versions. I had to update and reinstall and everything works fine now.

**Comment by kgilmer** [comment]
Hi @MaestroGlanz glad you found the source of the problem.  Yes I agree it would be nice to have a more foolproof installation, and the tabs are maybe too subtle.  I'd prefer to keep things concrete though as to not intimidate less technical users.  But, I cannot deny that your script is nice.  How would you derive `arch`?  Maybe parse `uname -a` output?  How about dealing w/ non supported variants?  (eg the script should fail gracefully if it's run on a system unsupported by regolith)

**Comment by Lucas-Monsegu** [comment]
Did a fresh install on ubuntu 22.10. Installed regolith using the recommanded way, then tried to install i3rocks-times and it failed.
```➜  ~ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.10
Release:	22.10
Codename:	kinetic
➜  ~ sudo apt install i3xrocks-time                
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 regolith-control-center : Depends: libgnome-desktop-3-19 (>= 3.33.4) but it is not installable
E: Unable to correct problems, you have held broken packages.
➜  ~ sudo apt install regolith-control-center
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 regolith-control-center : Depends: libgnome-desktop-3-19 (>= 3.33.4) but it is not installable
E: Unable to correct problems, you have held broken packages.
```
Added the time script manually to make it work.
